### PR TITLE
Migration to MCR

### PR DESCRIPTION
**Pull request by martiniadam**
We are publishing our images to MCR, and part of doing that is changing our pipelines to push to where MCR can pick it up. This PR:

- Changes the kustomize templates and pipelines to go to this webhook.

Still open question: 
- Are we changing staging portion to get the internal hook? Or should we keep it on mocimages?

### CODE REVIEW

File: templates/flavors/base/cluster-template.yaml [hunk 1]
```diff
@@ -326,7 +326,7 @@ spec:
           bind-address: "0.0.0.0"
           leader-elect-lease-duration: "60s"
           leader-elect-renew-deadline: "55s"
-      imageRepository: "ecpacr.azurecr.io"
```
Comment (zawachte) [inline templates/flavors/base/cluster-template.yaml:329]: this is unrelated. Please revert.

File: templates/flavors/mgmt/mgmt-machine.yaml [hunk 2]
```diff
@@ -52,7 +52,7 @@ spec:
         leader-elect-lease-duration: 60s
         leader-elect-renew-deadline: 55s
         terminated-pod-gc-threshold: "10"
-    imageRepository: ecpacr.azurecr.io
```
Comment (zawachte) [inline templates/flavors/mgmt/mgmt-machine.yaml:55]: this is unrelated. Please revert.

Comment (martiniadam) [reply templates/flavors/mgmt/mgmt-machine.yaml]: reverted

File: templates/flavors/base/cluster-template.yaml (hunk 1 above)
Comment (martiniadam) [reply templates/flavors/base/cluster-template.yaml]: reverted
